osu.Game.Rulesets.Taiko/Difficulty/TaikoLegacyScoreSimulator.cs

```diff
@@ -159,8 +156,7 @@ private void simulateHit(HitObject hitObject)
             {
                 int oldScoreIncrease = scoreIncrease;
 
-                // ReSharper disable once PossibleLossOfFraction (intentional to match osu-stable...)
-                scoreIncrease += (int)(scoreIncrease / 35 * 2 * (difficultyPeppyStars + 1) * modMultiplier) * (Math.Min(100, combo) / 10);
```

Comment:
Was this change intentional? It looks like it's undoing something specifically put in place to match expectations.

Reply:
Also, while we're at it, why is `modMultiplier` just *gone* from this line, too?

Reply:
Unless I'm missing something, as far as I can tell, the only thing that's changed is the removal of `modMultiplier`? This is explained by removing the mod multiplier as I mentioned in the description.

I didn't explicitly mention it, but I thought that was a given, given the entire purpose here is to remove any mod multipliers.

The same is done for osu!, just via removal of it being multiplied into `scoreMultiplier` (https://github.com/ppy/osu/pull/24779/files#diff-cb9b51e6336d258f91476bf755934d4e2e1593f9b697cfa7ea7a05f6886b4e0fL76-R66).

Reply:
Okay, fair point on the multiplier disappearing, but the other thing that's changed is that the `35` factor became a `35f`, therefore implicitly performing a promotion to float, while it was previously *intentionally* doing integer division with a comment suppression. So that is something that may have changed values unintentionally.

Reply:
Maybe `ILegacyScoreSimulator.Simulate`'s xmldoc should be updated to mention that it doesn't apply any mod multipliers and that it should be done in post? Probably as a remark.

Reply:
Hmm okay I was the impression that the `double` being there would cause an implicit conversion of int-to-double before the operations.

Reply:
> the `double` being there would cause an implicit conversion of int-to-double before the operations

It would, but my point is that that is probably undesirable. For reference:

![image](https://github.com/ppy/osu/assets/20418176/b3e21e46-27a4-41c2-9c9a-4ccb5055553b)

https://github.com/peppy/osu-stable-reference/blob/1531237b63392e82c003c712faa028406073aa8f/osu!/GameModes/Play/Rulesets/Ruleset.cs#L1093-L1098

Reply:
I mean that the double was there as part of `modMultiplier` previously, so I was trying to preserve behaviour, but that is incorrect.